age-plugin.md

```diff
@@ -216,6 +216,16 @@ Three commands are defined for this phase:
   wrap all the file keys to.
   - `IDENTITY` is the Bech32 encoding of an identity.
 - `(wrap-file-key; FILE_KEY)` - a file key to be wrapped.
+- `(extension-labels)` - indicates that the client is using "labels" to constrain
+  the recipient stanzas that it will accept.
+  - A label is defined in ABNF by the `argument` type from the [v1 age header][].
+  - Clients MUST check that all recipient stanzas wrapping a given file key have
+    the exact same label set. Clients MUST NOT permit partial overlapping sets.
+  - If the plugin does not respond with a `labels` command in phase 2, clients MUST
+    treat the plugin's recipient stanzas has having an empty label set.
+  - The `x25519` recipient stanza has an implicit empty label set.
+  - The `scrypt` recipient stanza has an implicit label set containing a single
+    random label.
```

Comment:
In https://github.com/str4d/rage/pull/365#issuecomment-1666825131 I noted that we also need to decide on how file header grease (currently unspecified) interacts with the label set, because what it results in is extra stanzas.

The simplest approach, and the one compatible with the behaviour of pre-labels clients, is that grease has an implicit empty label set. This lets it be added alongside `x25519` and any pre-labels plugins, but not alongside `scrypt`. However, it means that plugins which start using labels would as a side-effect never have their headers greased. This might be fine (and in fact would probably be a nice side-effect for corp usage, where you probably don't want random extra stanzas), but would be annoying for e.g. postquantum-only headers.

Reply:
It's indeed a greasing loss, but probably worth it to keep the implementation general. Empty label set sounds right.